CMakeLists.txt

```diff
@@ -49,12 +57,25 @@ include_directories(extern/rapidjson/include)
 
 add_subdirectory(src)
 
-find_package(Boost 1.50 REQUIRED COMPONENTS
-	program_options
-)
+# Tests
+file(GLOB TEST_SRCS tests/cpp/*.cpp)
```

Comment:
Please avoid using GLOB
see [Effective Modern CMake](https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1)
![image](https://user-images.githubusercontent.com/44251346/64066440-257c6180-cc4c-11e9-9a88-e1e1cdcd17c5.png)

Reply:
I insist on using GLOB in test CMake. Two reasons:
1. If someone adds a test file but forgets to modify test CMake. It must be an unavailable test in CI. CI is used for reducing rookie mistakes rather than enhancing the chances of it.
2. Only when users installed GTest, the test files will be compiled. There won't be a big problem, even though users forget to reload CMake. But, CMake will be loaded every time in CI, which guarantees all tests will be run every time.